regression/cbmc-library/memmove-01/constant.c

```diff
@@ -0,0 +1,50 @@
+#include <assert.h>
+#include <stdio.h>
+#include <string.h>
+
+void sort_items_by_criteria(int *item, int left, int right)
+{
+  int lidx = left, ridx = (left + right) / 2 + 1,
+      lsize = (left + right) / 2 - left + 1;
+  int tmp;
+
+  if(right - left < 1)
+    return;
+  sort_items_by_criteria(item, left, (left + right) / 2);
+  sort_items_by_criteria(item, (left + right) / 2 + 1, right);
+
+  while(ridx <= right && lidx < ridx)
+  {
+    if(item[lidx] > item[ridx])
+    {
+      tmp = item[ridx];
+      memmove(item + lidx + 1, item + lidx, lsize * sizeof(int));
+      item[lidx] = tmp;
+      ++ridx;
+      ++lsize;
+    }
+    ++lidx;
+    --lsize;
+  }
+}
+
+int main(int argc, char *argv[])
+{
+  int a[7] = {0};
+
+  //CBMC reports wrong results for 256, -2147221455, -2147221455, -2147221455, 16, -2147483600, 16384
```

Comment:
Is this still true, or was this the bug report that we're exhibiting we've fixed?

Reply:
Indeed, too naive copying. I've updated the comment as well as the original file (main.c in the same directory).